lookout/style/format/analyzer.py

```diff
@@ -166,10 +169,26 @@ def _load_config(cls, config: Mapping[str, Any]):
             "prune_dataset_ratio": .2,
             "n_estimators": 10,
             "random_state": 42,
-            "n_iter": 5
+            "n_iter": 5,
+            "line_length_limit": 500
         }
         final_config.update(config)
         # the incoming value can be a list from ASDF
         final_config["top_down_greedy_budget"] = TopDownGreedyBudget(
             *final_config["top_down_greedy_budget"])
         return final_config
+
+    @classmethod
+    def _filter_files(cls, files: Iterable[File], line_length_limit: int) -> Iterable[File]:
+        """
+        Filter files based on their maximum line length.
+
+        :param files: Files to filter.
+        :param line_length_limit: Maximum line length to accept a file.
+        :return: Files filtered.
+        """
+        for file in files:
+            text = file.content.decode("utf-8", "replace")
```

Comment:
We have https://docs.python.org/3.5/library/stdtypes.html#bytes.splitlines in 3.5+ so no need to decode

Reply:
`max(text.splitlines(), key=len)`

Reply:
So we sacrifice `rstrip()` and accurate utf-8 decoding which is not really needed here